dataprofiler/data_readers/csv_data.py

```diff
@@ -581,33 +581,32 @@ def is_match(cls, file_path, options=None):
         space_regex = data_utils.get_delimiter_regex(" ", quotechar)
 
         # Count the possible delimiters
-        with open(file_path, encoding=file_encoding) as f:
-            for line in f:
+        for line in data_as_str.split('\n'):
 
-                count = 0
-                line_count += 1
+            line_count += 1
+            count = 0
 
-                # Must have content in line, >1 due to the \n character
-                if len(line) <= 1:
-                    empty_line_count += 1
-                    continue
+            # Must have content in line, >1 due to the \n character
+            if len(line) <= 1:
+                empty_line_count += 1
+                continue
 
-                # Find the location(s) where each delimiter was detected
-                if delimiter:
-                    count = len([i.start() for i in re.finditer(delimiter_regex, line)])
-                else:                    
-                    # If no delimiter, see if spaces are regular intervals
-                    count = len([i.start() for i in re.finditer(space_regex, line)])
+            # Find the location(s) where each delimiter was detected
+            if delimiter:
+                count = len([i.start() for i in re.finditer(delimiter_regex, line)])
+            else:                    
+                # If no delimiter, see if spaces are regular intervals
+                count = len([i.start() for i in re.finditer(space_regex, line)])
```

Comment:
not saying change now, but maybe a cleanup issue, as it seems we already no if a delimiter exists above. we could remove the if statement and just do

```python
delimiter_regex = data_utils.get_delimiter_regex(" ", quotechar)
if delimiter:
    delimiter_regex = data_utils.get_delimiter_regex(delimiter, quotechar)
for ....
    count = len([i.start() for i in re.finditer(delimiter_regex, line)])
```